src/js/auth/ManagementClient.ts

```diff
@@ -116,3 +122,49 @@ export const extractUpdatableMetadataFromProfile = ({ name, nick, bio, website,
 export const sendEmailVerification = async (userId: string): Promise<void> => {
   await auth0ManagementClient.sendEmailVerification({ user_id: userId, client_id: clientId })
 }
+
+/**
+ * Retrieves roles the user is assigned. Returns at most 50 roles.
+ * @param userId Auth0 internal user id. Ex: auth0|234879238023482995
+ */
+export const getUserRoles = async (userId: string): Promise<Role[]> => {
+  return await auth0ManagementClient.getUserRoles({ id: userId, page: 0, per_page: 50 })
+}
+
+/**
+ * Sets roles for the user, making multiple calls to Auth0 since there is no consolidated set call.
+ * @param userId Auth0 internal user id. Ex: auth0|234879238023482995
+ * @param roles Array of role names (Ex: 'editor'), not Auth0 role IDs (Ex: 'rol_ds239fjdsfsd')
+ */
+export const setUserRoles = async (userId: string, roles: string[]): Promise<void> => {
```

Comment:
Kinda unfortunate that we have to make three API calls to set roles. Challenge is that Auth0 only has `assignRolesToUser` which strictly add roles and `removeRolesFromUser` which strictly removes roles. So we need at least two API calls.

The `getRoles` call, I'm using to pull role IDs. In theory we could hardcode this as a set of constants, but the risk is that they get out of sync. 

I could also do a `Promise.all` for the assign and remove calls to try to run them concurrently. Lmk if you'd like that.

Reply:
I think it's fine making 2 separate awaits